src/features/defi/components/Deposit/PairDepositWithAllocation.tsx

```diff
@@ -0,0 +1,320 @@
+import { Button, Stack, useColorModeValue } from '@chakra-ui/react'
+import type { Asset } from '@shapeshiftoss/asset-service'
+import type { AccountId } from '@shapeshiftoss/caip/dist/accountId/accountId'
+import type { MarketData } from '@shapeshiftoss/types'
+import get from 'lodash/get'
+import { calculateYearlyYield } from 'plugins/cosmos/components/modals/Staking/StakingCommon'
+import type { ControllerProps } from 'react-hook-form'
+import { useController, useForm, useWatch } from 'react-hook-form'
+import { useTranslate } from 'react-polyglot'
+import type { AccountDropdownProps } from 'components/AccountDropdown/AccountDropdown'
+import { Amount } from 'components/Amount/Amount'
+import { AssetIcon } from 'components/AssetIcon'
+import { Allocation } from 'components/DeFi/components/Allocation'
+import { AssetInput } from 'components/DeFi/components/AssetInput'
+import { FormField } from 'components/DeFi/components/FormField'
+import { Row } from 'components/Row/Row'
+import { Text } from 'components/Text'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+
+import { PairIcons } from '../PairIcons/PairIcons'
+
+type DepositProps = {
+  accountId?: AccountId | undefined
+  asset1: Asset
+  asset2: Asset
```

Comment:
Since this consistently deals with pairs, can we pass a fixed-size tuple of AssetIDs and select the Asset 1 and 2?